Bottom: 3e3ed8764dacca8c524408ad81171fb24c2360fc
Top:    7361d7acbe14aa60c745602ed7a31b345e8a1080
Author: scott snyder <sss@karma>
Date:   2020-05-27 21:39:22 -0400

Refresh of cmake

---

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e4d130c9..a130f0c56 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -470,8 +470,8 @@ else()
       if(DEFINED __tmp_LIBDIR)
         string(REPLACE "${NEO__GMM_INCLUDEDIR}" "${GMM_DIR}/include/igdgmm" NEO__GMM_INCLUDE_DIRS "${NEO__GMM_INCLUDE_DIRS}")
         string(REPLACE "${NEO__GMM_LIBDIR}" "${GMM_DIR}/${__tmp_LIBDIR}" NEO__GMM_LIBDIR "${NEO__GMM_LIBDIR}")
-        set(NEO__GMM_LIBRARY_PATH "${NEO__GMM_LIBDIR}")
       endif()
+      set(NEO__GMM_LIBRARY_PATH "${NEO__GMM_LIBDIR}")
 
       set(GMM_TARGET_NAME "igfx_gmmumd_dll")
       set(GMM_LINK_NAME ${NEO__GMM_LIBRARIES})
diff --git a/opencl/source/CMakeLists.txt b/opencl/source/CMakeLists.txt
index 3959f3d38..99cf109ac 100644
--- a/opencl/source/CMakeLists.txt
+++ b/opencl/source/CMakeLists.txt
@@ -171,6 +171,8 @@ if(${GENERATE_EXECUTABLE})
     )
   else()
     target_link_libraries(${NEO_DYNAMIC_LIB_NAME} ${GMM_LINK_NAME})
+    target_link_directories(${NEO_DYNAMIC_LIB_NAME} PUBLIC ${NEO__GMM_LIBRARY_PATH})
+    
     target_include_directories(${NEO_DYNAMIC_LIB_NAME} PRIVATE
       ${CMAKE_CURRENT_SOURCE_DIR}/dll/linux/devices${BRANCH_DIR_SUFFIX}
     )
diff --git a/package.cmake b/package.cmake
index 85dbb3561..ad43a0347 100644
--- a/package.cmake
+++ b/package.cmake
@@ -32,6 +32,7 @@ if(UNIX)
     else()
 	    set(OCL_ICD_VENDORDIR "/etc/OpenCL/vendors")
     endif()
+    set(OCL_ICD_VENDORDIR "${CMAKE_INSTALL_PREFIX}/etc/OpenCL/vendors")
   endif()
 
   set(OCL_ICD_RUNTIME_NAME ${CMAKE_SHARED_LIBRARY_PREFIX}${NEO_DLL_NAME_BASE}${CMAKE_SHARED_LIBRARY_SUFFIX})
